docs/node/signals.md

```diff
@@ -29,9 +29,9 @@ You invoke a Signal with `workflow.signal.signalName(...args)`. In the above cas
 
 ```ts
 const client = new WorkflowClient();
-const workflow = client.stub<Blocked>('blocked', { taskQueue: 'test' });
+const workflow = client.stub(unblockWithSignal, { taskQueue: 'test' });
```

Comment:
if task queue is basically always required in practice, should we do this instead? more ergonomic?

```suggestion
const workflow = client.stub({ 
	type: unblockWithSignal, // required
	taskQueue: 'test'        // practically required
});
```

also wondering if it read easier to have our users say `new Stub(...)` instead of `client.stub(...)`? pros and cons?

Reply:
I think I like the single-arg object.

> also wondering if it read easier to have our users say new Stub(...) instead of client.stub(...)? pros and cons?

You need the client info, right? So `new client.Stub({ })` or `new Stub({ client, ... })`?

Reply:
We need the client, so the constructor would have to take a `client` argument as @lorensr pointed out.
I don't love the name `stub` either but I'm not sure what's the best alternative.
We already have a `WorkflowClient`, `WorkflowHandle` maybe? Just `Workflow` (we already have a `Workflow` interface so maybe not the best choice)?

I also prefer the method over a constructor, it's a bit more convenient IMO, also 1 less import.
I don't have a strong preference for single vs 2 args.

Suggested alternatives:

```ts
client.workflow({ type: unblockWithSignal, taskQueue: 'test' })
client.stub({ type: unblockWithSignal, taskQueue: 'test' })
client.handle({ type: unblockWithSignal, taskQueue: 'test' })
client.workflow(unblockWithSignal, { taskQueue: 'test' })

```

Reply:
On the React theme, we could call them Refs 😄 

Similar to my Activities feedback, I like methods with verbs, so I can better understand what's happening by reading. These seem similarly convenient and understandable to me:

```js
client.createWorkflow({ type: unblockWithSignal, taskQueue: 'test' })
client.getWorkflow({ type: unblockWithSignal, taskQueue: 'test' })
client.getWorkflowRef({ type: unblockWithSignal, taskQueue: 'test' })

new client.Workflow({ type: unblockWithSignal, taskQueue: 'test' })
new client.WorkflowRef({ type: unblockWithSignal, taskQueue: 'test' })
```

Reply:
That name has actually cross my mind, definitely an option.

I changed the node PR to use the same methods as Java and PHP for lack of better ideas.

`client.newWorkflowStub` `newChildWorkflowStub` `newExternalWorkflowStub`

Reply:
```
client.handle
client.createWorkflowHandle
client.createHandle
client.newWorkflowHandle
client.newHandle
```

we like explicit because

```
client.createActivity
client.createWorkflow
client.createChildWorkflow
client.createExternalWorkflow
```